File: app/controllers/api/v2/players_controller.rb
class Api::V2::PlayersController < ApplicationController
  def index
    @players = Player.all  
  end

  def create
    @player = Player.create(
                                first_name: params[:first_name],
                                last_name: params[:last_name],
                                position: params[:position],
                                birthdate: params[:birthdate],
                                pro_team: params[:pro_team],
                                salary: params[:salary],
                                location: params[:location],
                                quote: params[:quote]
                                )
    render :show
  end

  def update
    @player = Player.find(params[:id])
    @player.update(
                    first_name: params[:first_name],
                    last_name: params[:last_name],
                    position: params[:position],
                    birthdate: params[:birthdate],
                    pro_team: params[:pro_team],
                    salary: params[:salary],
                    location: params[:location],
                    quote: params[:quote]
                  )
    render :show
  end

  def show
    @player = Player.find(params[:id])
  end

  def destroy
    @player = Player.find(params[:id])
    @player.destroy
    render json: {message: "Successfully Deleted Player"}
  end
end

File: app/assets/javascripts/players_controller.js
(function() {
  "use strict";

  angular.module("app").controller("playersController", function($scope, $http) {
    $scope.setup = function() {
      $http.get('api/v1/players.json').then(function(response) {
        $scope.players = response.data;
      });
    
    };

    $scope.addPlayer = function(newFirstName, newLastName, newPosition, newBirthday, newProTeam, newSalary, newLocation, newQuote) {
      var player = {
                      first_name: newFirstName,
                      last_name: newLastName,
                      position: newPosition,
                      birthdate: newBirthday,
                      pro_team: newProTeam,
                      salary: newSalary,
                      location: newLocation,
                      quote: newQuote
                  };
      
      $http.post('/api/v1/players.json', player).then(function(response) {
        $scope.players.push(response.data);
        $scope.newPlayerFirstName = null;
        $scope.newPlayerLastName = null;
        $scope.newPlayerPosition = null;
        $scope.newPlayerBirthday = null;
        $scope.newPlayerProTeam = null;
        $scope.newPlayerSalary = null;
        $scope.newPlayerLocation = null;
        $scope.newPlayerQuote = null;
      }, function(errors) {
        $scope.errors = errors.data;
      });

    };

    window.scope = $scope
  });
}());